happytransformer/happy_generation.py

```diff
@@ -0,0 +1,177 @@
+"""
+Contains the HappyGeneration class
+"""
+from dataclasses import dataclass
+from transformers import AutoModelForCausalLM
+from happytransformer.happy_transformer import HappyTransformer
+from happytransformer.toc.trainer import TOCTrainer
+from happytransformer.adaptors import get_adaptor
+
+default_greedy_settings = {
+    "do_sample": False,
+    "early_stopping": False,
+    "num_beams": 1,
+    "temperature": 0.65,
+    "top_k": 50,
+    "top_p": 1.0,
+    "repetition_penalty": 1,
+    "length_penalty": 1,
+    "no_repeat_ngram_size": 2,
+    'bad_words_ids': None,
+}
```

Comment:
Temperature is unused for greedy sampling. Why is it in here?

Reply:
Good point. I'll remove redundant parameters. I'll also remove the code to check for missing parameters.